opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java

```diff
@@ -186,9 +201,165 @@ private JSONObject createFeatureCollection() {
         return featureCollection;
     }
 
-    private ListTaskContract.Presenter getPresenter() {
-        return (ListTaskContract.Presenter) presenterCallBack;
+    public void saveJsonForm(String json) {
+        try {
+            JSONObject jsonForm = new JSONObject(json);
+            String encounterType = jsonForm.getString(ENCOUNTER_TYPE);
+            if (SPRAY_EVENT.equals(encounterType) || MOSQUITO_COLLECTION_EVENT.equals(encounterType) || LARVAL_DIPPING_EVENT.equals(encounterType)) {
+                saveMosquitoInterventionForm(jsonForm);
+            } else if (REGISTER_STRUCTURE_EVENT.equals(encounterType)) {
+                saveRegisterStructureForm(jsonForm);
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Error saving Json Form data", e);
+        }
     }
 
 
+    private org.smartregister.domain.db.Event saveEvent(JSONObject jsonForm, String encounterType, String bindType) throws JSONException {
```

Comment:
Remove this method, use the parent Method on BaseInteractor

Reply:
Done.